src/com/carolinarollergirls/scoreboard/core/impl/ScoreBoardImpl.java

```diff
@@ -82,6 +85,24 @@ protected void setupScoreBoard() {
     protected Object computeValue(PermanentProperty prop, Object value, Object last, Flag flag) {
         if (prop == Value.UPCOMING_JAM && !(value instanceof Jam)) {
             value = new JamImpl(this, getCurrentPeriod().getCurrentJam());
+        } else if (prop == Value.NO_MORE_JAM) {
```

Comment:
The logic isn't quite right here, if I stop on 30 it goes red rather than allowing another jam.

Reply:
There are exactly 30s between the moment when the display flips to 30 and the moment when the display flips to 0. So if you stop when the 30 is already showing, there are only 29.x seconds remaining and there should not be another Jam.

But I've changed the code to give you another Jam when you hit exactly 30.0s.